azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADUserGroupsPropertyValidator.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.autoconfigure.aad;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+/**
+ * Custom validator to validate that One of the two UserGroup properties are not empty
+ */
+public class AADUserGroupsPropertyValidator
```

Comment:
The validator annotation way is great, but wonder how about move this validation into the `AADAuthenticationProperties` using a `@PostConstruct` validation? Without introducing a new validator which is only used once in the `AADAuthenticationProperties` class.

Reply:
ahhh yes, I've actually used that in the past for simple validations. Will refactor